src/webnn_native/openvino/GraphIE.cpp

```diff
@@ -521,6 +521,101 @@ namespace webnn_native { namespace ie {
         return {};
     }
 
+    MaybeError Graph::AddGru(const op::Gru* gru) {
+        int n = 0;
+        auto inputs = gru->Inputs();
+        auto options = gru->GetOptions();
+        // TNC <=> NTC
+        std::vector<int64_t> order = std::vector<int64_t>{1, 0, 2};
+        const ngraph_node_t* orderNode =
+            AddConstantWithGraph<int64_t>(precision_e::I64, {order.size()}, order);
+        auto inputNode = const_cast<ngraph_node_t*>(mGraphNodeMap[inputs[n].Get()]);
+        n++;
+        ngraph_node_t* inputTransposeNode = nullptr;
+        IEStatusCode status = ngraph_transpose(inputNode, orderNode, &inputTransposeNode);
+        if (status != IEStatusCode::OK) {
+            dawn::ErrorLog() << "Failed to transpose gru input layout";
+        }
+        auto weightNode = const_cast<ngraph_node_t*>(mGraphNodeMap[inputs[n].Get()]);
+        n++;
+        auto recurrentWeightNode = const_cast<ngraph_node_t*>(mGraphNodeMap[inputs[n].Get()]);
+        n++;
+
+        dimensions_t shape;
+        ngraph_get_shape(inputTransposeNode, &shape);
+        auto batchSize = shape.dims[0];
+        ngraph_get_shape(weightNode, &shape);
+        auto numDirections = shape.dims[0];
+        auto steps = gru->GetSteps();
+        std::vector<size_t> stepsShape{batchSize};
+        std::vector<size_t> stepsData(batchSize, steps);
+        const ngraph_node_t* stepsNode =
+            AddConstantWithGraph<size_t>(precision_e::U64, stepsShape, stepsData);
+        auto hiddenSize = gru->GetHiddenSize();
+        ngraph_node_t* biasNode = nullptr;
+        if (options->bias != nullptr) {
```

Comment:
How about starting to use n and n++ here just for options? For example: int n = 3;

Reply:
Optimized.